gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -487,6 +494,38 @@ private static void setupConjurePythonProject(
         }
     }
 
+    private static void setupConjureRustProject(
+            Project project, Supplier<GeneratorOptions> options, Task compileConjure, Task compileIrTask) {
+        String rustProjectName = project.getName() + "-rust";
+        if (project.findProject(rustProjectName) != null) {
+            Configuration conjureRustConfig = project.getConfigurations().maybeCreate(CONJURE_RUST);
+
+            project.project(rustProjectName, subproj -> {
+                applyDependencyForIdeTasks(subproj, compileConjure);
+                File conjureRustDir = new File(project.getBuildDir(), CONJURE_RUST);
+                project.getDependencies().add(CONJURE_RUST, CONJURE_RUST_BINARY);
+                ExtractExecutableTask extractConjureRustTask = ExtractExecutableTask.createExtractTask(
+                        project, "extractConjureRust", conjureRustConfig, conjureRustDir, "conjure-rust");
+                project.getTasks().create("compileConjureRust",
```

Comment:
Out of curiosity, how are users going to consume the generated crate? What does publishing look like?

Reply:
The current support is designed for rust-based services, where your main crate would just depend on the generated Conjure crate via the local filesystem path.

Publishing support is a follow-up since it'll require a bit more work (we'll need to download a Cargo binary appropriate for the host, for example).